server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { delay } from "@fluidframework/common-utils";
+import { ILogger } from "./lambdas";
+
+export async function runWithRetry<T>(
+    api: () => Promise<T>,
+    callName: string,
+    maxRetries: number,
+    logger?: ILogger,
+    shouldRetry?: (error) => boolean,
+): Promise<T | undefined> {
+    let result: T | undefined;
+    let retryCount = 0;
+    const retryAfterMs = 1000;
```

Comment:
Should we make 'retryAfterMs' a parameter as well? That way caller can choose the interval as well?